src/app/features/reference-data/reference-data-amend/reference-data-amend.component.ts

```diff
@@ -35,13 +35,17 @@ export class ReferenceDataAmendComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
-    this.route.params.pipe(first()).subscribe(params => {
+    this.route.parent?.params.pipe(first()).subscribe(params => {
       this.type = params['type'];
+      // load the reference data admin type
+      this.referenceDataService.loadReferenceDataByKey(ReferenceDataResourceType.ReferenceDataAdminType, this.type);
+    });
+
+    this.route.params.pipe(first()).subscribe(params => {
       this.key = decodeURIComponent(params['key']);
```

Comment:
Does the parent route params contain the `key` as well?  I'm wondering if it's possible to combine these two very similar if statements to get the `type` and `key` from the same `params`

Reply:
You're presented with a 404 when trying to get them both from the parent route

Reply:
Can we get them both from the current (child) route, or are we happy with this approach?